cmake/GinkgoConfig.cmake.in

```diff
@@ -92,21 +92,17 @@ set(GINKGO_HAVE_HWLOC @GINKGO_HAVE_HWLOC@)
 set(GINKGO_HAVE_ROCTX @GINKGO_HAVE_ROCTX@)
 
 # Ginkgo installation configuration
-set(GINKGO_CONFIG_FILE_PATH "${CMAKE_CURRENT_LIST_DIR}")
-string(REPLACE "@GINKGO_INSTALL_CONFIG_DIR@" "" GINKGO_INSTALL_PREFIX "${GINKGO_CONFIG_FILE_PATH}")
-set(GINKGO_INSTALL_INCLUDE_DIR "${GINKGO_INSTALL_PREFIX}/@GINKGO_INSTALL_INCLUDE_DIR@")
-set(GINKGO_INSTALL_LIBRARY_DIR "${GINKGO_INSTALL_PREFIX}/@GINKGO_INSTALL_LIBRARY_DIR@")
-set(GINKGO_INSTALL_PKGCONFIG_DIR "${GINKGO_INSTALL_PREFIX}/@GINKGO_INSTALL_PKGCONFIG_DIR@")
-set(GINKGO_INSTALL_CONFIG_DIR "${GINKGO_INSTALL_PREFIX}/@GINKGO_INSTALL_CONFIG_DIR@")
-set(GINKGO_INSTALL_MODULE_DIR "${GINKGO_INSTALL_PREFIX}/@GINKGO_INSTALL_MODULE_DIR@")
+set(GINKGO_INSTALL_PREFIX "@PACKAGE_CMAKE_INSTALL_PREFIX@")
+set(GINKGO_INSTALL_INCLUDE_DIR "@PACKAGE_CMAKE_INSTALL_FULL_INCLUDEDIR@")
+set(GINKGO_INSTALL_LIBRARY_DIR "@PACKAGE_CMAKE_INSTALL_FULL_LIBDIR@")
 set(GINKGO_INSTALL_RPATH_FOR_HIP "-Wl,-rpath,${GINKGO_INSTALL_LIBRARY_DIR}")
 set(GINKGO_INSTALL_RPATH ${GINKGO_INSTALL_RPATH})
 set(GINKGO_INSTALL_RPATH_USE_ORIGIN ${GINKGO_INSTALL_RPATH_USE_ORIGIN})
 set(GINKGO_INSTALL_RPATH_DEPENDENCIES ${GINKGO_INSTALL_RPATH_DEPENDENCIES})
 
 # Forward Ginkgo's MODULE PATH and the PREFIX PATH for HIP and more
-list(APPEND CMAKE_MODULE_PATH "@CMAKE_MODULE_PATH@" "${GINKGO_INSTALL_MODULE_DIR}")
-list(APPEND CMAKE_PREFIX_PATH "@CMAKE_PREFIX_PATH@")
+list(APPEND CMAKE_MODULE_PATH "@PACKAGE_GINKGO_INSTALL_MODULE_DIR@")
```

Comment:
yes, true. This might be an issue at some point if we include transitive dependencies via CMake (e.g. Ginkgo includes Kokkos, Kokkos adds its own CMake modules that are not installed in the global module path), but I think that should be handled on the package manager level instead of by us, and is also a pretty obscure setting.

Reply:
I added the CMAKE_MODULE_PATH for HIP, which we set manually, to the GinkgoConfig file. I guess a good guideline is to forward only the module paths we add ourselves?

Reply:
I am not sure about the HIP because HIP cmake stuff does not work consistently